runtime/ms_rest_azure/lib/ms_rest_azure/resource.rb

```diff
@@ -2,14 +2,13 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 
+require_relative 'sub_resource'
+
 module MsRestAzure
   #
   # Class which represents any Azure resource.
   #
-  class Resource
-
-    # @return [String] the id of the resource.
-    attr_accessor :id
+  class Resource < SubResource

```

Comment:
Just curious why not use full namespace while inhering to avoid conflicts like `class Resource < MsRestAzure::SubResource`. That should also eliminate need for `require_relative 'sub_resource'`.

Moreover the name `Sub_Resource` inherited into `Resource` doesn't make sense as developer. I'd suggest renaming it as you mentioned that `SubResource` is not at all Azure concept so we better avoid explicitly referring it as SubResource.